tikv-client/src/main/java/com/pingcap/tikv/predicates/TiKVScanAnalyzer.java

```diff
@@ -318,8 +319,12 @@ TiKVScanPlan buildIndexScan(
         // -INF
         startKey = RowKey.createMin(id);
       } else {
-        // Comparison with null should be filtered since it yields unknown always
-        startKey = RowKey.toRowKey(id, r.lowerEndpoint());
+        try {
+          // Comparison with null should be filtered since it yields unknown always
+          startKey = RowKey.toRowKey(id, r.lowerEndpoint());
+        }catch (TiExpressionException e){
+          startKey = RowKey.createMin(id);
```

Comment:
just taking the min or max value will lead to an error answer.
here are the problems I found: insert a row with id = 9223372036854775808
1. `where id>0` can't get this row
2. `where id<0` get this row
3. `where id<=0` throw exception `Handle overflow for Long MAX` (maybe another bug)

Reply:
This is due to another bug because unsigned long PK in TiDB actually has two separate ranges when it is stored.
Unsigned long PK created by TiDB is treated as int64 rather than uint64.
Thus, the PK data is stored in order from `[0, INT64MAX]`, and then from `[INT64_MIN, 0)`.
e.g., 
`id <= 0` should be interpreted as range `[0, 0], [INT64_MIN, 0)`

Reply:
@shiyuhang0  sorry for the late response...

as @birdstorm  describe , `RowKey` with Long.Min and Long.Max range can fetch all rowset . I'll add some UTs later. 

How to resolve other derived errors is subject to further discussion.

Reply:
https://github.com/pingcap/tidb/blob/master/distsql/request_builder.go#L447-L510
You can check how TIDB handles this situation.

Reply:
I found this question involves not only this bug, I will work on it later.